packages/dds/tree/src/simple-tree/toMapTree.ts

```diff
@@ -248,13 +242,13 @@ function leafToMapTree(
 }
 
 /**
- * Checks an incoming value to ensure it is compatible with our serialization format.
+ * Checks an incoming {@link TreeLeafValue} to ensure it is compatible with its requirements.
  * For unsupported values with a schema-compatible replacement, return the replacement value.
  * For unsupported values without a schema-compatible replacement, throw.
  * For supported values, return the input.
  */
 function mapValueWithFallbacks(
-	value: TreeValue,
+	value: TreeLeafValue,
```

Comment:
Might as well use the public type for this, and leverage its public facing nice docs (which also needed fixing).